web/modules/custom/itk_cookie_message/src/Form/ItkCookieMessageConfigForm.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * @file
+ * Configuration form for the Cookie message module.
+ */
+
+namespace Drupal\itk_cookie_message\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageInterface;
+
+/**
+ * Configuration form for the Cookie message module.
+ */
+class ItkCookieMessageConfigForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'itk_cookie_message_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return array('itk_cookie_message.settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $settings = _itk_cookie_message_get_settings();
+
+    $form['tabs'] = array(
+      '#type' => 'vertical_tabs',
+      '#weight' => 99,
+    );
+
+    $form['general'] = array(
+      '#type' => 'details',
+      '#title' => $this->t('General settings'),
+      '#group' => 'tabs',
+    );
+
+    $key = 'cookie_name';
+    $form['general'][$this->getKey($key)] = array(
+      '#type' => 'textfield',
+      '#title' => $this->t('Cookie name'),
+      '#required' => TRUE,
+      '#default_value' => $settings[$key],
+    );
+
+    $key = 'cookie_lifetime';
+    $form['general'][$this->getKey($key)] = array(
+      '#type' => 'select',
+      '#title' => $this->t('Cookie lifetime'),
+      '#options' => array(
+        1 * 30 * 24 * 60 * 60 => $this->t('One month'),
+        1 * 365 * 24 * 60 * 60 => $this->t('One year'),
+      ),
+      '#default_value' => $settings[$key],
+    );
+
+    $languages = \Drupal::languageManager()->getLanguages();
+
+    foreach ($languages as $language) {
+      $language_id = $language->getId();
+      $settings = _itk_cookie_message_get_settings($language);
+
+      $form[$language_id] = array(
+        '#type' => 'details',
+        '#title' => $this->t($language->getName()),
+        '#group' => 'tabs',
+      );
+
+      $key = 'text';
+      $form[$language_id][$this->getKey($key, $language)] = array(
+        '#type' => 'textfield',
+        '#title' => $this->t('Message'),
+        '#description' => $this->t('Message to display in the dialog show to the users.'),
+        '#required' => TRUE,
+        '#default_value' => $settings[$key],
+      );
+
+      $key = 'read_more_url';
+      $form[$language_id][$this->getKey($key, $language)] = array(
+        '#type' => 'textfield',
+        '#title' => $this->t('Read more url'),
+        '#default_value' => $settings[$key],
+      );
+
+      $key = 'read_more_text';
+      $form[$language_id][$this->getKey($key, $language)] = array(
+        '#type' => 'textfield',
+        '#title' => $this->t('Read more text'),
+        '#default_value' => $settings[$key],
+      );
+
+      $key = 'accept_button_text';
```

Comment:
This key pattern is not very drupal link form building style? Normally you just write the string in the two places its used.